tests/installation/bootloader_hyperv.pm

```diff
@@ -144,13 +144,20 @@ sub run {
     my $ps = 'powershell -Command';
 
     my ($winserver, $winver, $vmguid);
+    my @win2k12r2 = ("6.3.9600");
+    my @win2k16   = ("10.0.14393", "10.0.16299");
+    my @win2k19   = ("10.0.17763");
 
-    $winver = $svirt->get_cmd_output("cmd /C ver");
-    if (grep { /Microsoft Windows \[Version 6.3.*\]/ } $winver) {
-        $winserver = 2012;
+    $winver = $svirt->get_cmd_output("$ps (Get-WmiObject Win32_OperatingSystem).Version");
+    $winver =~ s/^\s+|\s+$//g;
+    if (grep /^$winver$/, @win2k12r2) {
+        $winserver = '2012R2';
     }
-    elsif (grep { /Microsoft Windows \[Version 10.0.*\]/ } $winver) {
-        $winserver = 2016;
```

Comment:
I checked code change again. It seems actually there is no different handling for 2016 and 2019. Only need to differentiate 2012(6.3.*) and 2016+(10.0.*). I really do not recommend explict listing way, which has high chance needing to be extended once new hyperv deployments come.
What about '''
    if (grep { /Microsoft Windows \[Version 6.3.*\]/ } $winver) {
        $winserver = 2012;
    }
    elsif (grep { /Microsoft Windows \[Version 10.0.*\]/ } $winver) {
        $winserver = 2016_or_2019;
    }
'''

and change later 2016 checkings accordingly?

Reply:
OK, agree. Will update accordingly.